app/test/e2e/webshot-screenshot-post.spec.js

```diff
@@ -0,0 +1,110 @@
+            stubPuppeteer(sinonSandbox);
+            stubKoaSend(sinonSandbox, true);
+
+            requester = await getTestServer();
+
+            const response = await requester.get(`/api/v1/webshot?url=http://www.example.com&filename=newname&format=png`).send();
+            response.status.should.equal(200);
+            response.type.should.equal('application/png');
+        });
+    });
+
+    it('If there is no URL returns 400 Internal Server Error', async () => {
+        stubPuppeteer(sinonSandbox);
+        stubKoaSend(sinonSandbox, true);
+
+        requester = await getTestServer();
+
+        const response = await requester.get(`/api/v1/webshot?url=&filename=newname`).send();
+        response.status.should.equal(400);
+        response.body.should.have.property('errors').and.be.an('array');
+        response.body.errors[0].should.have.property('status').and.equal(400);
+        response.body.errors[0].should.have.property('detail').and.include('url param is required');
+    });
+
+    it('If there is no filename returns 400 Internal Server Error', async () => {
+        stubPuppeteer(sinonSandbox);
+        stubKoaSend(sinonSandbox, true);
+
+        requester = await getTestServer();
+
+        const response = await requester.get(`/api/v1/webshot?url=http://www.example.com&filename=`).send();
+        response.status.should.equal(400);
+        response.body.should.have.property('errors').and.be.an('array');
+        response.body.errors[0].should.have.property('status').and.equal(400);
+        response.body.errors[0].should.have.property('detail').and.include('filename param is required');
+    });
+
+    it('If format is invalid returns 400 Internal Server Error', async () => {
+        stubPuppeteer(sinonSandbox);
+        stubKoaSend(sinonSandbox, true);
+
+        requester = await getTestServer();
+
+        const response = await requester.get(`/api/v1/webshot?url=http://www.example.com&filename=newname&format=bmp`).send();
+        response.status.should.equal(400);
+        response.body.should.have.property('errors').and.be.an('array');
+        response.body.errors[0].should.have.property('status').and.equal(400);
+        response.body.errors[0].should.have.property('detail').and.include('format param is invalid');
+    });
+
+    it('If protocol in url is invalid returns 400 Internal Server Error', async () => {
+        stubPuppeteer(sinonSandbox);
+        stubKoaSend(sinonSandbox, true);
+
+        requester = await getTestServer();
+
+        const response = await requester.get(`/api/v1/webshot?url=file://www.example.com&filename=newname`).send();
+        response.status.should.equal(400);
+        response.body.should.have.property('errors').and.be.an('array');
+        response.body.errors[0].should.have.property('status').and.equal(400);
+        response.body.errors[0].should.have.property('detail').and.include('The protocol in url param is not valid. Use http or https.');
+    });
```

Comment:
suggestion: add a test scenario where the url is valid but returns a 404 if you try to open it - use nock to mock that request